packages/manager/src/features/PlacementGroups/PlacementGroupsDetail/PlacementGroupsDetail.tsx

```diff
@@ -0,0 +1,118 @@
+import { useHistory, useParams } from 'react-router-dom';
+
+import { CircleProgress } from 'src/components/CircleProgress';
+import { DocumentTitleSegment } from 'src/components/DocumentTitle';
+import { ErrorState } from 'src/components/ErrorState/ErrorState';
+import { LandingHeader } from 'src/components/LandingHeader';
+import { NotFound } from 'src/components/NotFound';
+import { SafeTabPanel } from 'src/components/Tabs/SafeTabPanel';
+import { TabLinkList } from 'src/components/Tabs/TabLinkList';
+import { TabPanels } from 'src/components/Tabs/TabPanels';
+import { Tabs } from 'src/components/Tabs/Tabs';
+import { useFlags } from 'src/hooks/useFlags';
+import {
+  useMutatePlacementGroup,
+  usePlacementGroupQuery,
+} from 'src/queries/placementGroups';
+import { getErrorStringOrDefault } from 'src/utilities/errorUtils';
+
+import { getAffinityLabel, getPlacementGroupLinodeCount } from '../utils';
+
+export const PlacementGroupsDetail = () => {
+  const flags = useFlags();
+  const { id, tab } = useParams<{ id: string; tab?: string }>();
+  const history = useHistory();
+  const placementGroupId = Number(id);
+  const {
+    data: placementGroup,
+    error: placementGroupError,
+    isLoading,
+  } = usePlacementGroupQuery(placementGroupId, Boolean(flags.vmPlacement));
+  const {
+    error: updatePlacementGroupError,
+    mutateAsync: updatePlacementGroup,
+    reset,
+  } = useMutatePlacementGroup(placementGroupId);
+  const errorText = getErrorStringOrDefault(updatePlacementGroupError ?? '');
+  const [isEditingLabel, setIsEditingLabel] = React.useState<boolean>(false);
+
+  if (!placementGroup) {
+    return <NotFound />;
+  }
+
+  if (placementGroupError) {
+    return (
+      <ErrorState errorText="There was a problem retrieving your Placement Group. Please try again." />
+    );
+  }
+
+  if (isLoading) {
+    return <CircleProgress />;
+  }
+
+  const linodeCount = getPlacementGroupLinodeCount(placementGroup);
+  const tabs = [
+    {
+      routeName: `/placement-groups/${id}`,
+      title: 'Summary',
+    },
+    {
+      routeName: `/placement-groups/${id}/linodes`,
+      title: `Linodes (${linodeCount})`,
+    },
+  ];
+  const { affinity_type, label } = placementGroup;
+  const affinityLabel = getAffinityLabel(affinity_type);
+  const tabIndex = tab ? tabs.findIndex((t) => t.routeName.endsWith(tab)) : -1;
+
+  const resetEditableLabel = () => {
+    setIsEditingLabel(false);
+
+    return `${label} (${affinityLabel})`;
+  };
+
+  const handleLabelEdit = (newLabel: string) => {
+    setIsEditingLabel(false);
+
+    if (updatePlacementGroupError) {
+      reset();
+    }
+
+    return updatePlacementGroup({ label: newLabel });
+  };
+
+  return (
+    <>
+      <DocumentTitleSegment segment={label} />
+      <LandingHeader
+        breadcrumbProps={{
+          onEditHandlers: {
+            editableTextTitle: isEditingLabel
+              ? label
+              : `${label} (${affinityLabel})`,
```

Comment:
Suggestion: instead of storing `isEditingLabel` in this component, the `editableTextTitle` prop can accept an optional function (i.e., it would have type `editableTextTitle: string | (isEditinglabel: boolean) => string`), which avoids duplicating state that is already tracked in the `EditableText` component as `isEditing`.

```suggestion
            editableTextTitle: (isEditingLabel) => isEditingLabel
              ? label
              : `${label} (${affinityLabel})`,
```

Reply:
Yeah that's fair, though the solution above isn't too much of an improvement because of how the initial state of this component is managed in the first place.

I went the route of adding a new prop instead which bypasses relying on state all together. See https://github.com/linode/manager/pull/10096/commits/31c2f92e4d05dde774b66d749ef0aff55ae98b05